bionlp/preprocess/dataset_preprocess.py

```diff
@@ -11,30 +11,47 @@
 from bionlp.data.dataset import Dataset as Dataset
 
 
-def encode_data_format(documents, raw_text, umls_params):
+def encode_data_format(documents, raw_text, umls_params, sentence_limit=0, label_blacklist=[]):
     logger.info('Encoding dataset into data format')
     document_dict = {did_: (dtext_, metamap_)
                      for (did_, dtext_, metamap_) in raw_text}
     documentList = []
+    total_sent_count = 0
+    sent_limit_reached = False
     for did, document in tqdm(documents):
         sentenceList = []
-        for sid, sent in enumerate(document):
+        sid = 0
+        for sent in document:
+            sent_is_blacklisted = False
             tid = 0
             tokenList = []
             for token in sent:
                 newToken = Token(token[0], tid)
-                newToken.attr['Annotation'] = token[2]
-                newToken.attr['offset'] = token[1]
-                newToken.attr['document'] = did
-                tokenList.append(newToken)
-                tid += 1
-            newSentence = Sentence(tokenList, sid)
-            sentenceList.append(newSentence)
+                label = token[2]
+                if label in label_blacklist:
+                    sent_is_blacklisted = True
+                    break
+                else:
+                    newToken.attr['Annotation'] = label
```

Comment:
Lets insantiate newToken via `newToken = Token(token[0], offset=token[1], document=did, Annotation=token[2])`. This requires to adjust the signature of `bionlp.data.token` though